src/k2/transport/Payload.cpp

```diff
@@ -36,7 +36,7 @@ Payload::PayloadPosition::PayloadPosition(_Size bIdx, _Size bOff, size_t offset)
     // Empty payload will have position of (bidx=0, boff=0, off=0)
 }
 
-Payload::Payload(BinaryAllocatorFunctor allocator):
+Payload::Payload(std::shared_ptr<BinaryAllocator> allocator):
```

Comment:
Can we pass the allocator by move instead of a shared pointer?

Reply:
I chose to use a shared pointer here to give users the flexibility to either use an individual allocator for a payload or use a shared allocator for multiple payloads. If we have the use case to use the same allocator for many payloads, the shared pointer is more efficient so that we don't have to create an allocator instance for each payload.

Run test_k23si_tpcc.sh locally and compare it with the master, didn't see any performance degradation. The times are almost the same.

Reply:
I see. If we want to keep the shared pointer we can change it to a seastar::lw_shared_ptr which is faster. The restrictions on it are that there is no polymorphism and it stays on the same core, which we should meet for this use case.

Reply:
I see, will change to use seastar::lw_shared_ptr then.

Reply:
done